src/main/webapp/app/overview/exercise-details/exercise-details-student-actions.component.ts

```diff
@@ -141,6 +143,42 @@ export class ExerciseDetailsStudentActionsComponent {
             });
     }
 
+    isManualFeedbackRequestsAllowed(): boolean {
+        return this.exercise.allowManualFeedbackRequests ?? false;
+    }
+
+    private feedbackSent = false;
+
+    isFeedbackRequestButtonDisabled(): boolean {
+        const participation = this.studentParticipation ?? (this.exercise.studentParticipations && this.exercise.studentParticipations.first());
+
+        const showUngradedResults = true;
+        const latestResult = participation?.results && participation.results.find(({ rated }) => showUngradedResults || rated === true);
+        const allHiddenTestsPassed = latestResult?.score !== undefined && latestResult.score >= 100;
+
+        const requestAlreadySent = (participation?.individualDueDate && participation.individualDueDate.isBefore(Date.now())) ?? false;
+
+        return !allHiddenTestsPassed || requestAlreadySent || this.feedbackSent;
+    }
+
+    requestFeedback() {
+        this.exercise.loading = true;
+        this.courseExerciseService
+            .requestFeedback(this.exercise.id!)
+            .pipe(finalize(() => (this.exercise.loading = false)))
+            .subscribe({
+                next: (participation: StudentParticipation) => {
+                    if (participation) {
+                        this.feedbackSent = true;
```

Comment:
It seems you receive the student participation from this request. Above, you already check (based on the participation) if the request is already sent. Couldn't you therefore update the participation/the `individualDueDate` instead of introducing a new local helper variable?

Reply:
no that didnt work